### PR TITLE
Dark room support

### PR DESCRIPTION
**Pull request by SteveRussell33**

### CODE REVIEW

File: src/memorypad.cpp [hunk 1]
```diff
@@ -253,20 +253,20 @@ struct MemoryPadTrackpad : OpaqueWidget {
     }
   }
 
-  void draw(const DrawArgs& args) override {
+  void drawLayer(const DrawArgs& args, int layer) override {
+    if (layer != 1 || !module) return;
```
Comment (epadilla) [inline src/memorypad.cpp]: Stylistically, I'd prefer to have braces if thats ok:
```
if (layer != 1 || !module) {
  return;
}
```

File: src/memorypad.cpp [hunk 2]
```diff
@@ -253,20 +253,20 @@ struct MemoryPadTrackpad : OpaqueWidget {
...
+
     OpaqueWidget::draw(args);
```
Comment (epadilla) [inline src/memorypad.cpp]: This should be removed right? (Also, I wasn't sure if it should go at the start or end. Is it not an issue to have the call to super never happen in the case then the if you added in ln 257 returns early?)

File: src/memorypad.cpp [hunk 3]
```diff
@@ -298,6 +298,8 @@ struct MemoryPadTrackpad : OpaqueWidget {
     nvgCircle(args.vg, puckDot.x, puckDot.y, 8.0);
     nvgFillColor(args.vg, SCHEME_YELLOW);
     nvgFill(args.vg);
+
+    Widget::drawLayer(args, layer);
```
Comment (epadilla) [inline src/memorypad.cpp:302]: Should we not be calling the direct superclass? (`OpaqueWidget::drawLayer(args, layer);`)

**Review by epadilla — CHANGES_REQUESTED** [review]
Thank you so much for making implementing this! It was something I wanted to do but hadn't looked into it. Def looks so much better.  I have a some questions and suggestion, if thats ok. 

On a separate note, I noticed that the light bloom effect doesnt happen automatically for the outputs jack (and I do see it happening for the VCV modules). I'll look into this later...

**Comment by epadilla** [comment]
Actually, nvm on my comment about bloom. The light bloom is visible on the outputs, it was just kinda subtle.

File: src/memorypad.cpp (hunk 1 above)
Comment (SteveRussell33) [reply src/memorypad.cpp]: Done

File: src/memorypad.cpp (hunk 2 above)
Comment (SteveRussell33) [reply src/memorypad.cpp]: Yeah, my mistake. I'll delete that line, it's fine.

File: src/memorypad.cpp (hunk 3 above)
Comment (SteveRussell33) [reply src/memorypad.cpp:302]: ``OpaqueWidget`` is derived from ``Widget``
https://vcvrack.com/docs-v2/structrack_1_1widget_1_1OpaqueWidget
https://vcvrack.com/manual/Migrate2#3-5-Draw-custom-light-like-widgets-at-full-brightness

Comment (epadilla) [reply src/memorypad.cpp:302]: Perhaps this betrays my more limited C++ knowledge, but I thought one should call the direct super.

**Review by epadilla — APPROVED** [review]
Thanks for amending!

**Pull request merged**